mnemosyne/core/beam.py

```
@@ -7920,6 +7925,50 @@ def reclaim_orphans(
+        result = {"gists_orphans": 0, "memory_embeddings_orphans": 0, "gists_deleted": 0, "memory_embeddings_deleted": 0}
+
+        # Check if child tables exist
+        tables_present = {row["name"] for row in cursor.execute(
+            "SELECT name FROM sqlite_master WHERE type='table' AND name IN ('gists','memory_embeddings')"
+        )}
+
+        for child, id_col in [("gists", "memory_id"), ("memory_embeddings", "memory_id")]:
+            if child not in tables_present:
+                continue
+            # Count orphans
+            cursor.execute(f"SELECT COUNT(*) FROM {child} WHERE {id_col} IS NOT NULL AND {id_col} NOT IN ({parent_union})")
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)